lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt

```diff
@@ -270,82 +270,33 @@ class GatewayProvider(
     }
 
     private fun getEstimateFee(
-        payload: InvokeTransactionPayload,
+        payload: TransactionPayload,
         blockId: BlockId,
     ): Request<EstimateFeeResponse> {
         val url = feederGatewayRequestUrl("estimate_fee")
-        val body = serializeInvokeTransactionPayload(payload)
+        val body = when (payload) {
+            is InvokeTransactionPayload -> serializeInvokeTransactionPayload(payload)
+            is DeployAccountTransactionPayload -> serializeDeployAccountTransactionPayload(payload)
+            is DeclareTransactionPayload -> Json.encodeToJsonElement(DeclareTransactionPayloadSerializer, payload).jsonObject
```

Comment:
Why do we need to call a function for invoke and deploy and declare is just encoded using serializer?

Reply:
First two are defined in GatewayProvider, and all they do is basically build json object specific to gateway. 
For declare, there is a serializer defined in serializers that works for both rpc and gateway, so in order to not have a boiler plate and reuse that serializer we need to call it a bit differently.

Reply:
Okay, makes sense.